pkg/translator/prometheus/normalize_name_test.go

```diff
@@ -146,6 +149,47 @@ func TestOtelReceivers(t *testing.T) {
 
 }
 
+func TestTrimPromSuffixes(t *testing.T) {
+	registry := featuregate.NewRegistry()
+	require.NoError(t, registry.RegisterID(normalizeNameGateID, featuregate.StageBeta))
+
+	normalizer := NewNormalizer(registry)
+
+	assert.Equal(t, "active_directory_ds_replication_network_io", normalizer.TrimPromSuffixes("active_directory_ds_replication_network_io_bytes_total", pmetric.MetricTypeSum, "bytes"))
+	assert.Equal(t, "active_directory_ds_replication_sync_object_pending", normalizer.TrimPromSuffixes("active_directory_ds_replication_sync_object_pending_total", pmetric.MetricTypeSum, "{objects}"))
```

Comment:
I don't believe {objects} is a valid OM unit.  From the ABNF: https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#abnf, it can include ALPHA, DIGIT, _, or : characters.  For units that are a "count of X" (this is a "count of objects"), prometheus considers those unitless.

This also would mean that `connections`, `{requests}`, `{scanned}/{returned}`, etc should be unitless.

Reply:
Got it, but in case of the prometheus receiver receiving such a metric_name/unit pair, should we just completely ignore it?

Should we built a map of "valid" units that we shall trim?

Reply:
I think the prometheus server will already reject `{`/`}` characters, but in general, we dont need to be more restrictive than the prometheus server is.  We shouldnt build a map of valid units or anything.  Maybe add a comment for those cases saying that it isn't valid OM, but we test it for completeness.